src/lib/scanner.ts

```diff
@@ -21,13 +30,15 @@ export class ScanRunner {
      */
     public async run(targets: string[]): Promise<RuleResult[]> {
         // Create the arg array.
-        const args: string[] = await this.createPathlessArgArray(targets);
+        const args: Inputs = this.createPathlessArgArray(targets);
 
         // Invoke the scanner.
-        const executionResult: ExecutionResult = await this.invokeAnalyzer(args);
+        // const executionResult = await this.invokeAnalyzer(args);
+        const executionResult: AnyJson = await this.invokeAnalyzerWithoutCli(args);
 
         // Process the results.
-        return this.processPathlessResults(executionResult);
+        // return this.processPathlessResults(executionResult); 
+        return executionResult as RuleResult[];
```

Comment:
It seems odd that this run method returns a RuleResult array but the runDfa method returns a string. Why are they different?

And how is it possible that the invokeAnalyzerWithoutCli output can be casted to a RuleResult array and the other to a string? That seems odd to me.